website/docs/language/import/generating-configuration.mdx

```diff
@@ -0,0 +1,123 @@
+
+```hcl
+provider "aws" {
+  region = "eu-west-1"
+}
+
+import {
+  to = aws_iot_thing.bar
+  id = "foo"
+}
+```
+
+### 2. Plan the import
+Run `terraform plan -generate-config-out=generated.tf`. You should see output like the following:
+
+
+```
+$ terraform plan
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+```
+
+### 3. Review the generated code
+
+Open `generated.tf` in your editor and review the generated code. It should look like this:
+
+```hcl
+resource aws_iot_thing "bar" {
+  name = "foo"
+}
+```
+
+Add or remove resource arguments as desired. Now is an appropriate time to add variables or references to other resources.
+
+### 4. Apply the import
+Run `terraform apply`. You should see output like the following:
+
+```
+$ terraform apply
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+aws_iot_thing.bar: Importing... [id=foo]
+aws_iot_thing.bar: Import complete [id=foo]
+
+Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.
+
+```
+
+Make sure that your generated code, along with your other Terraform code, is added to your version control system.
+
+## Limitations
+
+Terraform generates configuration for importable resources during the plan by requesting values for resource attributes from the provider. For certain resources with complex schemas, it may not be possible for Terraform to construct a valid configuration from these values. In this case, you may see an error like the following when planning:
+
+```
+$ terraform plan -generate-config-out=generated.tf
+╷
+│ Error: Conflicting configuration arguments
+│ 
+│   with aws_instance.ubuntu,
+│   on g.tf line 20, in resource "aws_instance" "ubuntu":
+│   20:   ipv6_address_count                   = 0
+│ 
+│ "ipv6_address_count": conflicts with ipv6_addresses
+╵
+```
+
+In this case, Terraform still generates configuration and writes it to `generated.tf`. You will probably want to remove one or more of the `ipv6_address_count` and `ipv6_addresses` arguments.
```

Comment:
Are there any other errors folks might run into? The error here feels pretty clear, but I'm trying to think if we should highlight any weird edge cases?
```suggestion
In the example above, Terraform still generates configuration and writes it to `generated.tf`. This error stems from a conflict between the `ipv6_address_count` and `ipv6_addresses` arguments. You could fix the error by removing one of these two arguments.
```